src/plugins/console/public/lib/mappings/mappings.js

```diff
@@ -292,42 +303,52 @@ export function retrieveAutoCompleteInfo(settings, settingsToRetrieve) {
   const mappingPromise = retrieveSettings('fields', settingsToRetrieve);
   const aliasesPromise = retrieveSettings('indices', settingsToRetrieve);
   const templatesPromise = retrieveSettings('templates', settingsToRetrieve);
+  const indexTemplatesPromise = retrieveSettings('indexTemplates', {
+    ...settingsToRetrieve,
+    indexTemplates: settingsToRetrieve.templates,
+  });
 
-  $.when(mappingPromise, aliasesPromise, templatesPromise).done((mappings, aliases, templates) => {
-    let mappingsResponse;
-    if (mappings) {
-      const maxMappingSize = mappings[0].length > 10 * 1024 * 1024;
-      if (maxMappingSize) {
-        console.warn(
-          `Mapping size is larger than 10MB (${mappings[0].length / 1024 / 1024} MB). Ignoring...`
-        );
-        mappingsResponse = '[{}]';
-      } else {
-        mappingsResponse = mappings[0];
+  $.when(mappingPromise, aliasesPromise, templatesPromise, indexTemplatesPromise).done(
+    (mappings, aliases, templates, indexTemplates) => {
+      let mappingsResponse;
+      if (mappings) {
+        const maxMappingSize = mappings[0].length > 10 * 1024 * 1024;
```

Comment:
I know, this code is not yours, but, please, think about this code. It can fail, if `mappings` is an empty array.

And even more, this code is failing silently. We should wrap it up with `try/catch` and provide some notifying about the error. Could you, please, add some correct checks for this code to be safe during the execution? Thanks.